Use python3-rohmu library instead of pghoard

raphlopez opened the pull request:
The [rohmu project](https://github.com/aiven/rohmu) was recently split from its parent project, [pghoard](https://github.com/aiven/pghoard). As myhoard relies on rohmu for performing backup tasks (object storage upload/download, compression, encryption), the package dependency must be updated. 

Resolves aiven/rohmu issue [#7](https://github.com/aiven/rohmu/issues/7)

alanfranz-at-work reviewed (dismissed):
Test failures seem related.

rikonen commented:
Is there some reason why this adds the rohmu dependencies but does not remove references to pghoard?

raphlopez commented:
There are a couple references to pghoard outside of rohmu (`pghoard.common.set_stream_nonblocking`, `pghoard.common.increase_pipe_capacity`). 
They both seem to be helper methods and are not highly coupled with anything else in pghoard. If it makes sense to copy them over into this project and remove pghoard dependency entirely, I can definitely do that.

raphlopez commented:
Made [PR #48](https://github.com/aiven/rohmu/pull/48) in rohmu to duplicate pghoard utils. We'll need to wait for the merge and next rohmu release before this can be merged without any pghoard dependencies.

alexole commented:
PR was merged, could you update this PR please? I need it to get into aiven-core for my ticket @raphlopez

alanfranz-at-work commented on myhoard/version.py:
```diff
@@ -0,0 +1 @@
+__version__ = '1.0.6-45-g2ce5877'
```
This mustn't be committed. I don't know why this is not in .gitignore, though. I had already noticed and thought about this.

raphlopez replied:
Thanks for pointing that out. Removed the file and added it to the .gitignore.

alanfranz-at-work reviewed (changes requested):
LGTM beyond the version glitch.

raphlopez committed:
Use python3-rohmu library instead of pghoard

Since python3-rohmu is now packaged independently of
pghoard, update code references (and pkg dependencies)
to start using rohmu intead of pghoard for object storage
and backup compression/encryption

alexole commented on requirements.txt:
```diff
@@ -3,6 +3,6 @@ botocore
-pghoard >= 2.1.0
 PyMySQL >= 0.9.2
 PySocks
+rohmu
```
It won't work with any version of `rohmu`, would be better to specify `rohmu >= 1.0.2`

alexole commented on myhoard.spec:
```diff
@@ -18,14 +17,15 @@ BuildRequires:  python3-PyMySQL >= 0.9.2
 Requires:       python3-aiohttp
 Requires:       python3-cryptography >= 0.8
 Requires:       python3-PyMySQL >= 0.9.2
+Requires:       python3-rohmu
```
Same here and above (see my comment about the version)

alexole reviewed (approved):
LGTM
